onnxruntime/core/providers/coreml/builders/impl/pad_op_builder.cc

```diff
@@ -150,6 +150,17 @@ bool PadOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderInputParam
       LOGS(logger, VERBOSE) << "constant_value must be a constant initializer.";
       return false;
     }
+
+    int32_t constant_value_type;
+    if (!GetType(*input_defs[2], constant_value_type, logger)) {
+      LOGS(logger, VERBOSE) << "Get constant_value type failed";
+      return false;
+    }
```

Comment:
nit: an initializer always has a type as it's constant data, so this check may be unnecessary.